pkg/cmd/pipeline/start.go

```diff
@@ -289,7 +293,23 @@ func (opt *startOptions) startPipeline(pipelineStart *v1beta1.Pipeline) error {
 		if err != nil {
 			return err
 		}
-		pr.Spec.Timeout = &metav1.Duration{Duration: timeoutDuration}
+		pr.Spec.Timeouts.Pipeline = &metav1.Duration{Duration: timeoutDuration}
```

Comment:
@piyush-garg @vdemeester @pradeepitm12 @chmouel this will be a breaking change as folks using Pipelines version below 0.36 won't be able to use `--timeout` flag without enabling the feature flags. In order to make it backward compatible we need to take care of following:
- Check the installed Pipelines version
      - If the Pipeline version is greater than and equal to 0.36, proceed with the new timeouts
      - Else use the older one
 Now to check the Pipelines version we'll have to query the configmap to fetch the Pipelines version ie one more API call. There can be a case where the user has installed Tekton in a namespace which is not part of default namespaces (tekton-pipelines, openshift-pipelines) then in that case there will be a failure in querying the ConfigMap